components/ui-api-layer/internal/domain/k8s/pod_service.go

```diff
@@ -46,8 +53,48 @@ func (svc *podService) List(namespace string, pagingParams pager.PagingParams) (
 			return nil, fmt.Errorf("Incorrect item type: %T, should be: *Pod", item)
 		}
 
+		svc.ensureTypeMeta(pod)
+
 		pods = append(pods, pod)
 	}
 
 	return pods, nil
 }
+
+func (svc *podService) Update(name, namespace string, update v1.Pod) (*v1.Pod, error) {
+	podKind, podAPIVersion := svc.podTypeMeta()
+	if update.Kind != podKind || update.APIVersion != podAPIVersion {
+		return nil, errors.New("pod's kind or apiVersion should not be changed")
+	}
+
+	updated, err := svc.client.Pods(namespace).Update(&update)
+	if err != nil {
+		return nil, err
+	}
+
+	svc.ensureTypeMeta(updated)
+
+	return updated, nil
+}
+
+func (svc *podService) Delete(name, namespace string) error {
+	return svc.client.Pods(namespace).Delete(name, nil)
+}
+
+// Kubernetes API used by client-go doesn't provide kind and apiVersion so we have to add it here
+// See: https://github.com/kubernetes/kubernetes/issues/3030
+func (svc *podService) ensureTypeMeta(pod *v1.Pod) {
+	podKind, podAPIVersion := svc.podTypeMeta()
+	if pod.APIVersion == "" {
```

Comment:
Replace whole `TypeMeta` struct for Pod, so return it from `podTypeMeta` method.